pkg/version/semver/semver.go

```diff
@@ -0,0 +1,47 @@
+package semver
+
+import (
+	"regexp"
+	"strings"
+
+	"github.com/Masterminds/semver/v3"
+)
+
+var (
+	releasePattern       = regexp.MustCompile("^v[0-9]")
+	branchReleasePattern = regexp.MustCompile("^v[0-9].x")
+)
+
+type Version string
+
+// HasReleasePrefix validates the value against the "legacy" releasePattern
+// It is not necessarily a valid SemVer, but does have the expected "release prefix".
+func (v Version) HasReleasePrefix() bool {
+	return releasePattern.MatchString(string(v))
+}
+
+// HasBranchReleasePrefix validates the value as having a minor version set to `x`
+func (v Version) HasBranchReleasePrefix() bool {
+	return branchReleasePattern.MatchString(string(v))
+}
+
+// IsDevOrPrerelease reports if the version is a non-stable, dev, or pre-release build.
+func (v Version) IsDevOrPrerelease() bool {
+	if v == "dev" {
+		return true
+	}
+
+	semVersion, err := semver.NewVersion(string(v))
+	return err != nil || // When version is not SemVer it is dev
+		semVersion.Prerelease() != "" // When the version includes pre-release assume dev
+}
+
+// IsRC validates the version is SemVer with a "Prerelease" matching the "RC" variety
+func (v Version) IsRC() bool {
```

Comment:
This may be naive by why do we even have an explicit prerelease? Why do we want `-alphax` to behave differently than `-rcx`?

I know this isn't a change you have made, but this has always seemed like a code smell to me, I'd love it if, in the case that we absolutely need to treat alpha different than rc, we can document here exactly why that is (is it charts, KDM, etc.).

Reply:
This is totally a good callout for more docs. Since I don't recall of the top of my head I'll look over my notes and my investigation tabs a bit more on monday. I think I should be able to create us some docs that are helpful to give that context too. (If I had to guess I think UI related ones?)